src/panels/config/automation/action/ha-automation-action-row.ts

```diff
@@ -290,17 +301,56 @@ export default class HaAutomationActionRow extends LitElement {
   private _handleAction(ev: CustomEvent<ActionDetail>) {
     switch (ev.detail.index) {
       case 0:
-        this._switchYamlMode();
+        this._runAction();
         break;
       case 1:
-        fireEvent(this, "duplicate");
+        this._switchYamlMode();
         break;
       case 2:
+        fireEvent(this, "duplicate");
+        break;
+      case 3:
         this._onDelete();
         break;
     }
   }
 
+  private async _runAction() {
+    const validated = await validateConfig(this.hass, {
+      action: this.action,
+    });
+
+    if (!validated.action.valid) {
+      showAlertDialog(this, {
+        title: this.hass.localize(
+          "ui.panel.config.automation.editor.actions.invalid_action"
+        ),
+        text: validated.action.error,
+        warning: true,
```

Comment:
I don't thing this should be a warning? As it is not a destructive action?

Reply:
I tagged it a warning because the action didn't run because the config is invalid.

Is warning meant for destructive things only ? I guess we don't differentiate between error and warning.

Reply:
We don't have strict rules (we should) but it is used now for destructive things, and not for errors

Reply:
```suggestion
```